core/src/replay_stage.rs

```diff
@@ -9369,4 +9370,192 @@ pub(crate) mod tests {
         assert_eq!(working_bank.slot(), good_slot);
         assert_eq!(working_bank.parent_slot(), initial_slot);
     }
+
+    #[test]
+    #[should_panic(expected = "assertion `left == right` failed")]
```

Comment:
I get the intention here, but if an assertion incorrectly triggers, such as the one on line 9414, the test will still pass right ?

Reply:
Could potentially add a custom message to the `assert_eq` within `process_duplicate_confirmed_slots()` and then detect the custom string ?